src/main/scala/io/opentargets/etl/backend/cancerbiomarkers/CancerBiomarkers.scala

```diff
@@ -0,0 +1,168 @@
+      .join(efoIds, Seq("tumor_type"), "left_outer")
+      .drop("name")
+
+    // override specific genes and turn GeneId into ENSG ID
+    val biomarkerWithDiseaseAndGene = biomarkerWithDisease
+      .withColumn("Gene", explode(col("Gene")))
+      .withColumn("Gene",
+                  when(col("Gene") === "C15orf55", "NUTM1")
+                    .when(col("Gene") === "MLL", "KMT2a")
+                    .when(col("Gene") === "MLL2", "KMT2D")
+                    .otherwise(col("Gene")))
+      .withColumn("Gene", upper(col("Gene"))) // call upper for
+      .join(ensemblIdLookup, Seq("Gene"))
+      .drop("Gene")
+      .withColumn("uid", monotonically_increasing_id)
+
+    // pubmed references
+    val biomarkerWithDiseaseGenePubmedDF = biomarkerWithDiseaseAndGene
+      .withColumn("Source", array_distinct(col("Source")))
+      .withColumn("Source", explode(col("Source")))
+      .withColumn("pmid",
+                  when(col("Source").startsWith("PMID"),
+                       expr("substring(Source, 6,length(Source))").cast(LongType)))
+      .withColumn(
+        "clinicalTrial",
+        when(
+          col("Source").startsWith("NCT"),
+          struct(
+            col("Source") as "name",
+            lit("Clinical Trial") as "description",
+            concat(lit("https://clinicaltrials.gov/ct2/show/"), col("Source")) as "link"
+          )
+        )
+      )
+      .groupBy({
+        "clinicalTrial" +: biomarkerWithDiseaseAndGene.columns
+      }.map(col): _*)
+      .agg(collect_set("pmid") as "sourcesPubmed")
+
+    val otherSourcesDF = biomarkerWithDiseaseGenePubmedDF
+      .select(col("uid"), trim(regexp_extract(col("Source"), "[\\w ]+", 0)) as "source")
+      .join(sourceDF, Seq("source"), "left_outer")
+      .withColumn(
+        "otherReferences",
+        when(!(col("Source").startsWith("NCT") || col("Source").startsWith("PMID")),
+             struct(
+               col("source") as "name",
+               col("label") as "description",
+               col("url") as "link"
+             ))
+          .otherwise(null)
+      )
+      .drop("source", "label", "url")
+
+    val flatten_distinct = array_distinct _ compose flatten
+
+    val biomarkerDiseaseGeneSources = biomarkerWithDiseaseGenePubmedDF
+      .join(otherSourcesDF, Seq("uid"), "left_outer")
+      .withColumn("s", array("otherReferences", "clinicalTrial"))
+      .distinct
+      .groupBy("uid", "id", "associationType", "drugName", "evidenceLevel", "target", "disease")
+      .agg(flatten_distinct(collect_set("sourcesPubmed")) as "sourcesPubmed",
+           flatten_distinct(collect_set("s")) as "sourcesOther")
+      .drop("uid")
+
+    biomarkerDiseaseGeneSources.as[CancerBiomarker]
+  }
+
+  /** Return map on input IOResources */
+  private def getMappedInputs(config: Configuration.CancerBiomarkersSection)(
+      implicit sparkSession: SparkSession): Map[String, IOResource] = {
+
+    val inputs = config.inputs
+
+    val mappedInputs = Map(
```

Comment:
@JarrodBaker you dont need to do this. This is the whole point of having `IOResourceConfig` in the configuration section. You can pass this straight to the map and therefore to the `readFrom` like
```Scala
    val mappedInputs = Map(
      "biomarkers" -> inputs.biomarkers,
      "diseaseMapping" -> inputs.diseaseMapping,
      "sourceMapping" -> inputs.diseaseMapping,
      "targetMapping" -> inputs.targetMapping
    )
    IoHelpers
      .readFrom(mappedInputs)
```

Reply:
Thanks for spotting. Fixed!